src/server.rs

```diff
@@ -21,158 +79,491 @@ pub struct Server {
+                                        Ok(server_url) => {
+                                            info!("Server url: {server_url}");
+                                            url_sender.send_replace(Some(server_url));
+                                        }
+                                        Err(err) => error!(
+                                            "Error when passing {server_url} as server url: {err}"
+                                        ),
+                                    }
+                                }
+                            }
+                            Ok(None) => {
+                                // do nothing
+                            }
+                            Err(err) => error!("Error collecting Server logs: {err}"),
+                        }
+                    }
+                }
+                Err(err) => error!("{err}"),
+            }
+        });
+    }
+}
+
+impl Drop for Server {
+    fn drop(&mut self) {
+        match block_on(self.stop()) {
+            Ok(_) => {}
+            Err(err) => error!("Failed to stop server on Drop, reason: {err}"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ServerSettingsResponse {
+    pub values: SettingsValues,
+    pub base_url: Url,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SettingsValues {
+    pub server_version: String,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
 pub struct Config {
-    server: PathBuf,
+    /// Full `nodejs` binary path
+    ///
+    /// Includes the OS-dependent suffix:
+    /// - `linux` - `node`
+    /// - `macos` - `node`
+    /// - `windows` - `node.exe`
     node: PathBuf,
+    /// Full `ffmpeg` binary path
+    ///
+    /// - `linux` - `ffmpeg`
+    /// - `macos` - `ffmpeg`
+    /// - `windows` - `ffmpeg.exe`
     ffmpeg: PathBuf,
+    /// Full `ffprobe` binary path
+    ///
+    /// - `linux` - `ffprobe`
+    /// - `macos` - `ffprobe`
+    /// - `windows` - `ffprobe.exe`
     ffprobe: PathBuf,
+    /// server.js binary path
+    server: PathBuf,
+    /// Whether or not to set the `NO_CORS` environment variable to `1`
+    /// and disable the CORS checks in the Server.
+    no_cors: bool,
 }
 
 impl Config {
     /// Create a Config using the same directory for all binaries
     ///
+    /// The directory should contain the following binaries:
+    ///
+    /// - node(.exe) - depending on target OS being `windows` or not.
+    /// - ffmpeg(.exe) - depending on target OS being `windows` or not.
+    /// - server.js
+    ///
     /// # Errors
     ///
     /// When one of the binaries required for running the server is missing.
-    pub fn new(directory: PathBuf) -> Result<Self, Error> {
+    pub fn at_dir(directory: PathBuf, no_cors: bool) -> Result<Self, Error> {
         if directory.is_dir() {
+            let node = directory.join(Self::node_bin(None)?);
```

Comment:
This implementation seems convoluted compared to the one already present

Reply:
This makes sure to list all the missing binaries instead only the first one.
If a user complains about the Service not being able to run, this would help us with Support.